roscala/src/test/scala/coop/rchain/roscala/VmSpec.scala

```diff
@@ -619,4 +619,59 @@ class VmSpec extends FlatSpec with Matchers {
 
     rtnCtxt.rslt shouldBe Fixnum(12)
   }
+
+  "OpLookupToReg" should "lookup binding in continuation when compiler can't find the compile-time binding for method." in {
+
+    /**
+      * This opcode sequence will be generated by the RBL
+      * compiler with the expression: (cat 1)
+      *
+      * litvec:
+      * 0:	{RequestExpr}
+      * 1:	'cat
+      * codevec:
+      * 0:	alloc 1
+      * 1:	lit 1,arg[0]
+      * 2:	lookup 1,trgt
+      * 3:	xmit/nxt 1
+      *
+      *
+      * this use case will go through the `absent` branch of `OpLookUpReg`, and the successful use case should be:
+      *
+      * (defProc (sbo& Top) (cat x) (+ x 1))
+      * (free [cat] (cat 1))     // this will print out `2`
+      *
+      * (free [〈id〉 …] 〈body〉) ⇒ result of 〈body〉
+      * The free form informs the compiler that the 〈id〉 are to be treated as free in 〈body〉;
+      * otherwise, the compiler will issue warnings that the 〈id〉 do not have a compile-time binding.
+      * And then it will be found in somewhere.
+      *
+      */
+    val codevec = Seq(
+      OpAlloc(n = 1),
+      OpImmediateLitToArg(literal = `1`, arg = 0),
+      OpLookupToReg(lit = `1`, reg = trgt),
+      OpXmit(unwind = false, next = true, nargs = 1)
+    )
+
+    val rtnCtxt = Ctxt.outstanding(1)
+
+    val code = Code(litvec = Seq(Niv, RblString("cat")), codevec = codevec)
+
+    val ctxt        = Ctxt(code, rtnCtxt, LocRslt)
+    // todo we set RequestExpr's parent and meta to the most outer case for simplicity, actually there would be RequestExprSBO and RequestExprMeta
+    val requestExpr = new RequestExpr(Seq.empty[Ob])
+
+    // we need RBLtopenv to end the recurse `lookup`
+    requestExpr.parent = new RBLtopenv
+    requestExpr.meta = topMeta
```

Comment:
I just checked - in this example the `meta` of the `RequestExpr` is not `TopMeta`.
Just do `requestExpr.meta = Meta.empty` in order to not confuse our future selves.

Reply:
Done